sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1086,6 +1106,17 @@ public Read<K, V> withRedistributeNumKeys(int redistributeNumKeys) {
       return toBuilder().setRedistributeNumKeys(redistributeNumKeys).build();
     }
 
+    public Read<K, V> withOffsetDeduplication(boolean offsetDeduplication) {
+      /*
+       * TODO(tomstepp): Auto-enable offset deduplication if: redistributed and !allowDuplicates.
+       * Until then, enforce explicit enablement only with redistributed without duplicates.
+       */
+      checkState(
+          isRedistributed() && !isAllowDuplicates(),
```

Comment:
similarly throw this when building instead of here, so it's not sensitive to ordering of the builder methods being called

Reply:
Would this be best done in `KafkaIO.Read.expand()`? Looks like there is already a good amount of configuration checking done there. Or are you referring to somewhere else?

Reply:
Took a stab at it, please let me know if you had something else in mind.